rasa/nlu/components.py

```diff
@@ -322,6 +341,70 @@ def validate_required_components_from_data(
         )
 
 
+def warn_of_competing_extractors(pipeline: List["Component"]) -> None:
+    """Warns the user when using competing extractors.
+
+    Competing extractors are e.g. `CRFEntityExtractor` and `DIETClassifier`.
+    Both of these look for the same entities based on the same training data
+    leading to ambiguity in the results.
+    """
+    general_purpose_entity_extractors = [
+        "DIETClassifier",
+        "CRFEntityExtractor",
+        "MitieEntityExtractor",
+    ]
+    extractors_in_pipeline = find_components_in_pipeline(
+        general_purpose_entity_extractors, pipeline
+    )
+    if len(extractors_in_pipeline) > 1:
+        rasa.shared.utils.io.raise_warning(
+            f"You have defined multiple entity extractors that do the same job "
+            f"in your pipeline: "
+            f"{', '.join(extractors_in_pipeline)}. This can lead to the same entity getting "
+            f"extracted multiple times. Please read the documentation section "
+            f"on entity extractors to make sure you understand the implications: "
+            f"{DOCS_URL_COMPONENTS}#entity-extractors"
+        )
+
+
+def warn_of_competition_with_regex_extractor(
+    pipeline: List["Component"], data: TrainingData
+) -> None:
+    """Warns when regex entity extractor is competing with a general one.
+
+    This might be the case when the following conditions are all met:
```

Comment:
can we please also describe the input params to follow the google docstyle standards?